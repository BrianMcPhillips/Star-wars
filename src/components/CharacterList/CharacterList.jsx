import React from 'react';
import PropTypes from 'prop-types';
import Character from '../Character/Character';

const CharacterList = ({ data }) => {
  const characterStuff = data.map(character => {
    <li>
      <Character {...character}/>
    </li>;
  });

  return (
    <ul>
      {characterStuff}
    </ul>
  );
};

CharacterList.propTypes = {
  data: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired
  }))
};

export default CharacterList;
