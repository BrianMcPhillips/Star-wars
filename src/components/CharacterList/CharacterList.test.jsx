import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterList from './CharacterList';
import { MemoryRouter } from 'react-router-dom';

describe('CharacterList component', () => {
  afterEach(() => cleanup());
  it('renders CharacterList', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CharacterList data={[
          {
            name:'Luke Skywalker',
            hair_color:'blond',
            birth_year: '19BBY',
            url: 'https://swapi.dev/api/people/1/',
            films: [
              'https://swapi.dev/api/films/1/',
              'https://swapi.dev/api/films/2/',
              'https://swapi.dev/api/films/3/',
              'https://swapi.dev/api/films/6/'
            ],
            vehicles:[
              'https://swapi.dev/api/vehicles/14/',
              'https://swapi.dev/api/vehicles/30/'
            ],
            starships:[
              'https://swapi.dev/api/starships/12/',
              'https://swapi.dev/api/starships/22/'
            ]
          }
        ]}/>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
