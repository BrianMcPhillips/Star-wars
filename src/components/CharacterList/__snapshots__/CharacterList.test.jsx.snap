// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component renders CharacterList 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <h2>
        Luke Skywalker
      </h2>
    </li>
  </ul>
</DocumentFragment>
`;
